src/backend/joanie/core/models/products.py

```diff
@@ -1015,14 +1015,7 @@ def _get_schedule_dates(self):
         """
         Return the schedule dates for the order.
         """
-        try:
-            signed_contract_date = self.contract.student_signed_on
-        except Order.contract.RelatedObjectDoesNotExist as e:
-            logger.error(
-                "Contract does not exist, cannot retrieve withdrawal date",
-                extra={"context": {"order": self.to_dict()}},
-            )
-            raise ObjectDoesNotExist("Order has no contract") from e
+        signed_contract_date = timezone.now()
```

Comment:
I'll just rename this method or at least complete the doc string. Because I think a newcomer could understand it could use this method to get order schedule dates at any time but that's wrong, this method MUST be used only to generate the payment schedule and populate the installment field.